### PR TITLE
Fixes #1888: Default to PHP 7.1.

### LINKED ISSUE #1888: Provide Default Platform Configuration in Composer
Currently the composer configuration in BLT is based on the host machine. This can be problematic for newer systems (running PHP 7) or systems that rely on their VMs to match production / cloud configuration.

I would propose that we add default platform settings to the composer.required.json file so that BLT defaults all composer dependencies to PHP 5.6. See: https://getcomposer.org/doc/06-config.md#platform

```
"config": {
  "platform": {
    "php": "5.6"
  }
},
```

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

**Comment by grasmash** [comment]
@danepowell @geerlingguy @mikemadison13 Will this have any notable consequences that I should be aware of?

File: composer.required.json [hunk 1]
```diff
@@ -35,6 +35,11 @@
       "Acquia\\Blt\\Custom\\": "../../../blt/src/"
     }
   },
+  "config": {
+    "platform": {
+      "php": "7"
```
Comment (danepowell) [inline composer.required.json]: Won't this prevent BLT from being installed or run on host machines using PHP <7?

**Comment by danepowell** [comment]
We should also set up Pipelines to use PHP7:
````
version: 1.1.0
services:
  - mysql
  - php:
      version: 7.1
````

We should also make sure to include release notes indicating that the default versions have changed, and folks should re-initialize DrupalVM, Travis, and Pipelines to get the change locally.

**Comment by mikemadison13** [comment]
People will def. need to change back to 5.6 if not running 7. That will cause problems for sure.

File: composer.required.json (hunk 1 above)
Comment (dpagini) [inline composer.required.json]: Curious about this answer...? Also, if this is the general approach of BLT, can't we undo #1829 as well?

**Pull request merged**